src/api/vdom.js

```diff
@@ -145,17 +144,22 @@ function resolveTagName (name) {
     return name;
   }
 
-  // We try and return the cached tag name, if one exists.
-  if (name[$name]) {
-    return name[$name];
+  // We try and return the cached tag name, if one exists. This will work with
+  // *any* web component of any version that defines a `static is` property.
+  if (name.is) {
+    return name.is;
   }
 
-  // If it's a custom element, we get the tag name by constructing it and
-  // caching it.
+  // Get the name for the custom element by constructing it and using the
+  // localName property. Cache it and lookup the cached value for future calls.
   if (name.prototype instanceof HTMLElement) {
+    if (name[$name]) {
```

Comment:
I tried removing having to need to use the `$name` symbol, but decided it was best to keep because I think it's okay to set `is` on CEs passed to `define()`, but don't want to set it on *any* ol custom element as a way to cache the result just in case.